samples/bluetooth/alexa_gadget/src/gadgets/gadgets_profile.h

```diff
@@ -0,0 +1,182 @@
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Gadgets event types */
+enum gadgets_evt_type {
+	/** Handshake with peer completed */
+	GADGETS_EVT_READY,
+
+	/** SetAlert Directive */
+	GADGETS_EVT_SETALERT,
+
+	/** DeleteAlert Directive */
+	GADGETS_EVT_DELETEALERT,
+
+	/** SetIndicator Directive */
+	GADGETS_EVT_SETINDICATOR,
+
+	/** ClearIndicator Directive */
+	GADGETS_EVT_CLEARINDICATOR,
+
+	/** StateUpdate Directive */
+	GADGETS_EVT_STATEUPDATE,
+
+	/** Speechmarks Directive */
+	GADGETS_EVT_SPEECHMARKS,
+
+	/** Tempo Directive */
+	GADGETS_EVT_MUSICTEMPO,
+
+	/** Custom Directive */
+	GADGETS_EVT_CUSTOM,
+
+	/* Custom Event sent */
+	GADGETS_EVT_CUSTOM_SENT
+};
+
+/** @brief Gadgets event details */
+struct gadgets_evt {
+	enum gadgets_evt_type type;
+	union {
+		/* Note: these are the structs generated by protoc compiler,
+		 * based on .proto files.
+		 * Abstraction not added in order to reference original doc:
+		 * https://developer.amazon.com/en-US/docs/alexa/alexa-gadgets-toolkit/defined-interfaces.html
+		 */
+
+		/** Parameters for @ref GADGETS_EVT_SETINDICATOR */
+		struct _notifications_SetIndicatorDirectivePayloadProto *set_indicator;
+
+		/** Parameters for @ref GADGETS_EVT_SETALERT */
+		struct _alerts_SetAlertDirectivePayloadProto *set_alert;
+
+		/** Parameters for @ref GADGETS_EVT_DELETEALERT */
+		struct _alerts_DeleteAlertDirectivePayloadProto *delete_alert;
+
+		/** Parameters for @ref GADGETS_EVT_STATEUPDATE */
+		struct _alexaGadgetStateListener_StateUpdateDirectivePayloadProto *state_update;
+
+		/** Parameters for @ref GADGETS_EVT_SPEECHMARKS */
+		struct _alexaGadgetSpeechData_SpeechmarksDirectivePayloadProto *speech_marks;
+
+		/** Parameters for @ref GADGETS_EVT_MUSICTEMPO */
+		struct _alexaGadgetMusicData_TempoDirectivePayloadProto *music_tempo;
+
+		/** Parameters for @ref GADGETS_EVT_CUSTOM */
+		struct {
+			const char *name;
+			const void *payload;
+			size_t size;
+		} custom_directive;
+	} parameters;
+};
+
+/** @brief Callback type for gadget events. */
+typedef void (*gadgets_profile_cb_t)(const struct gadgets_evt *evt);
+
+/**@brief Initialize the Profile.
+ *
+ * @details Initializes profile state and the underlying
+ *          Gadgets Bluetooth Service.
+ *
+ * @param[in] evt_handler Event handler function pointer.
+ *
+ * @retval 0 If initialization is successful.
+ *           Otherwise, a negative value is returned.
+ */
+int gadgets_profile_init(gadgets_profile_cb_t evt_handler);
+
+/**@brief Start advertising using the Gadgets advertisement format.
+ *
+ * @param[in] bondable_type Set to true if the "new bond" payload format should
+ *            be used. False when reconnecting to bonded peer.
+ *
+ * @retval 0 If advertisement is successfully started.
+ *           Otherwise, a negative value is returned.
+ */
+int gadgets_profile_adv_start(bool bondable_type);
+
+/**@brief Send a custom event to Alexa using pre-configured Custom Event.
```

Comment:
```suggestion
/**@brief Send a custom event to Alexa using a preconfigured custom event.
```
No reason to capitalize 'custom event' , please change in all occurrences.